src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -332,9 +332,29 @@ public boolean isGeneratedNameSameAsOriginal() {
      * @return true if suggested filepath is same as existing filepath.
      */
     public boolean isGeneratedPathSameAsOriginal() {
-        Optional<Path> newDir = databaseContext.getFirstExistingFileDir(preferences.getFilePreferences());
+        FilePreferences filePreferences = preferences.getFilePreferences();
+        Optional<Path> newDir = databaseContext.getFirstExistingFileDir(filePreferences);
+        if (newDir.isEmpty()) {
+            // could not find default path
+            return false;
+        }
```

Comment:
Hi @k3KAW8Pnf7mkmdSMPHz27! 

Thank you for reviewing my PR. Could you elaborate on this request a bit more? You mentioned two directories are empty. Other than `linkedFile.findIn(databaseContext, preferences.getFilePreferences())`, what is the other one? 

I'm also trying to understand the logic here. For the part you quoted, it means if the user did not set default path in preference, this function would return `false`, which means the two options would be disabled. Similarly, in line 353-357, if`(currentDir.isEmpty())`, the function is also going to return `false`, under the logic that in this case, we are not able to find the current directory of the file. Are you expecting different behaviors for these two cases?

Thanks a lot!

Reply:
> Are you expecting different behaviors for these two cases?

Oups, sorry about both the delayed response and the typo in my comment, I've updated it.
The original comment is regarding changing the behavior of the method when neither is present (no default path and no linked file)

previously the only value returned from the method was this,

https://github.com/JabRef/jabref/blob/69467d3b955e34011c61bea6f51d4b237407067b/src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java#L346

and if both
* `linkedFile.findIn(databaseContext, preferences.getFilePreferences())`
* `databaseContext.getFirstExistingFileDir(preferences.getFilePreferences()`

are empty we return 

https://github.com/JabRef/jabref/blob/69467d3b955e34011c61bea6f51d4b237407067b/src/main/java/org/jabref/model/util/OptionalUtil.java#L25-L26

`true`.

> which means the two options would be disabled

wouldn't this mean that they are enabled since it gets negated later on?

https://github.com/JabRef/jabref/blob/69467d3b955e34011c61bea6f51d4b237407067b/src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java#L305

P.S, that would be an awesome test case :smiley: 

Huge thanks for addressing all the nitpicks and yet again, sorry about the delayed response. I'll try to be more responsive if you have follow up questions about this :/

Reply:
I realized I dodged your question a bit, the short(ish) answer to

> I'm also trying to understand the logic here.

is that I believe the separation of concern can be improved upon in the original method

https://github.com/JabRef/jabref/blob/69467d3b955e34011c61bea6f51d4b237407067b/src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java#L334

In my opinion, the "correct" method signature should be `public Optional<Boolean> isGeneratedPathSameAsOriginal() {`
because the two cases you bring up don't match very well with a `true` or `false` value to the question "is the generated path the same as the original path" (since there doesn't exist a generated path or an original path). That is part of what I hoped to bring up in the follow-up issue :stuck_out_tongue: 

In my opinion the check for 

> if the user did not set default path in preference

and

> Similarly, in line 351-354, `if(currentDir.isEmpty())`

provides improved separation of concern if they are done in

https://github.com/JabRef/jabref/blob/69467d3b955e34011c61bea6f51d4b237407067b/src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java#L302-L305

So that those lines read

* is it an online link?
* is there a linked file?
* can we generate a path?
* is the generated path the same as the current path?

Reply:
As I am writing this I realize that this is probably quite a bit into the nitpicking territory. 

Imo the main thing a PR should do is improve on the code, which your PR does, so view this as a nitpick, would you like to address it otherwise we merge this PR as-is as soon as you get back to us

Reply:
Addressing the nitpick could either be to change the method signature or to match the true/false behavior of the original method (in which case changing the method signature will be part of the follow-up issue when I have time to write it)

Reply:
Hi @k3KAW8Pnf7mkmdSMPHz27, I now apologize for my delayed response. Thanks a lot for the detailed explanation. I realized I confused myself a bit earlier. 

> For the part you quoted, it means if the user did not set default path in preference, this function would return false, which means the two options would be disabled.

This would actually be the opposite to what I previously said. Simply with `isGeneratedPathSameAsOriginal()` equal to `false` would mean the two options being enabled rather than disabled. 

I'm now getting what you would like to address here and I agree `true/false` is not a perfect answer to these two cases. 

My apology for not responding earlier, but since the PR has been merged, I would be happy to take a look at the follow-up issue when I have time. :)